node/network/collator-protocol/src/collator_side.rs

```diff
@@ -195,8 +195,24 @@ where
 
 	state.our_validators_groups.insert(relay_parent, our_validators.clone());
 
+	// We may be already connected to some of the validators. In that case,
+	// advertise a collation to them right away.
+	// To the validators that we are not connected to, issue a connection request.
+	let (advertise_to, connect_to): (Vec<_>, Vec<_>) = our_validators.into_iter()
+		.partition(|v| if let Some(peer) = state.known_validators.get(&v) {
+			state.peer_views.contains_key(peer)
+		} else {
+			false
+		});
+
+	for advertise_to in advertise_to.into_iter() {
+		if let Some(peer) = state.known_validators.get(&advertise_to).cloned() {
+			advertise_collation(ctx, state, relay_parent, vec![peer]).await?;
+		}
+	}
+
 	// Issue a discovery request for the validators of the current group and the next group.
-	connect_to_validators(ctx, relay_parent, state, our_validators).await?;
+	connect_to_validators(ctx, relay_parent, state, connect_to).await?;
```

Comment:
I think we'd still need to issue `connect_to_validators` call to the whole group, i.e. `our_validators`.
The reason for this is that we revoke previous request here, so in theory we might disconnect from the old peers we're still interested in (at least they could potentially be removed from the priority group).

Reply:
ValidatorDiscovery implements reference counting per peer, so that if a validator was requested in two requests, and one request was revoked, it will not be removed from the priority group.

Reply:
Ok, take a look at the logic and the test now, i think that it does that

Reply:
Hmm, now that I think about it, the downside of that approach is that we will re`Declare` to already connected validators this way, but I guess it's harmless?